barretenberg/ts/src/barretenberg_cmds/index.ts

```diff
@@ -0,0 +1,39 @@
+import { type Circuit, type FieldsAndBinary } from './types.js';
+
+export * from './types.js';
+
+/**
+ * Common commands exposed by native bb CLI, used for key generation, proving, verifying, etc.
+ */
+export interface BarretenbergCmds {
+  getProvingKey(circuit: Circuit): Promise<{ pk: FieldsAndBinary }>;
+
+  getVerificationKey(circuit: Circuit): Promise<{ vk: FieldsAndBinary }>;
+
+  prove(circuit: Circuit, witness: Buffer): Promise<{ proof: FieldsAndBinary; vk: FieldsAndBinary }>;
```

Comment:
Proper return types are unclear to me here. Looking at the bb native CLI, it seems like it's "free" to get the proof as fields and the vk from calling prove, but in the wasm implementation this requires additional calls. What should we shoot for, a more coarse or a more granular API?